File: Euler7.js
var sPrime = 10001;
var pset = [2];

function primeCheck(num) {
  for(var i = 2; i < num/2; i++){
    if(num % i === 0){
      return false;
    }
  }
  return true;
}

var q = 3;
function listxPrimes(given) {
  while(pset.length < given){
    if(primeCheck(q)){
      pset.push(q);
    }
    q += 2;
  }
}

listxPrimes(sPrime);

console.log(pset[sPrime-1]);

File: Euler7.py
sPrime = 10001
pset = [2]
q = 3

def primeCheck(num):
    for i in range(2, int(num/2)):
        if num % i == 0:
            return False
    return True

def listxPrimes(given):
    global q
    while(len(pset) < given):
        if primeCheck(q):
            pset.append(q)
        q+=2

listxPrimes(sPrime)
print(pset[sPrime-1])
